### PR TITLE
Package xtensor-zarr

### PR DESCRIPTION
**Pull request by davidbrochart**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by davidbrochart** [comment]
We need to release xtensor-io and xtensor-zarr before merging this PR.

**Comment by davidbrochart** [comment]
@SylvainCorlay @JohanMabille do you agree to be maintainers of this package?

**Comment by JohanMabille** [comment]
Yes

File: recipes/xtensor-zarr/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
```
Comment (Anthchirp) [inline recipes/xtensor-zarr/build.sh:2]:
```suggestion
set -e

```
just to make sure errors are caught

File: recipes/xtensor-zarr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+test:
+  commands:
+    - test -d ${PREFIX}/include/xtensor-zarr  # [unix]
+    - test -f ${PREFIX}/include/xtensor-zarr/xzarr_hierarchy.hpp  # [unix]
+    - test -f ${PREFIX}/lib/cmake/xtensor-zarr/xtensor-zarrConfig.cmake  # [unix]
+    - test -f ${PREFIX}/lib/cmake/xtensor-zarr/xtensor-zarrConfigVersion.cmake  # [unix]
+    - if exist %LIBRARY_PREFIX%\include\xtensor-zarr\xzarr_hierarchy.hpp (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\lib\cmake\xtensor-zarrConfig.cmake (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\lib\cmake\xtensor-zarrConfigVersion.cmake (exit 0) else (exit 1)  # [win]
```
Comment (Anthchirp) [inline recipes/xtensor-zarr/meta.yaml:40]:
```suggestion
    - if not exist %LIBRARY_PREFIX%\include\xtensor-zarr\xzarr_hierarchy.hpp (exit 1)  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\xtensor-zarrConfig.cmake (exit 1)  # [win]
    - if not exist %LIBRARY_PREFIX%\lib\cmake\xtensor-zarrConfigVersion.cmake (exit 1)  # [win]
```
maybe cleaner?

File: recipes/xtensor-zarr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+about:
+  home: http://quantstack.net/xtensor
```
Comment (Anthchirp) [inline recipes/xtensor-zarr/meta.yaml:43]: this is a 404 link?

File: recipes/xtensor-zarr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: Implementation of the Zarr version 3.0 core protocol based on xtensor
+  description: This library enables loading chunked arrays stored in the Zarr format, in various formats and file systems.
+  doc_url: http://xtensor-zarr.readthedocs.io
```
Comment (Anthchirp) [inline recipes/xtensor-zarr/meta.yaml:49]:
```suggestion
  doc_url: https://xtensor-zarr.readthedocs.io
```

File: recipes/xtensor-zarr/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+{% set sha256 = "51fb22c0701df3f9d41aa811db2dff034eb409fa6f83b35dfde9fe24fd26fec5" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/xtensor-stack/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (Anthchirp) [inline recipes/xtensor-zarr/meta.yaml:12]:
```suggestion

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://github.com/xtensor-stack/{{ name }}/archive/{{ version }}.tar.gz
  sha256: 51fb22c0701df3f9d41aa811db2dff034eb409fa6f83b35dfde9fe24fd26fec5
```
This is slightly cleaner. I don't think we advertise the older style any more?

**Review by Anthchirp — COMMENTED** [review]
overall looks sensible

File: recipes/xtensor-zarr/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  license_file: LICENSE
+  summary: Implementation of the Zarr version 3.0 core protocol based on xtensor
+  description: This library enables loading chunked arrays stored in the Zarr format, in various formats and file systems.
+  doc_url: http://xtensor-zarr.readthedocs.io
+  dev_url: https://github.com/xtensor-stack/xtensor-zarr
+
+extra:
+  recipe-maintainers:
+    - davidbrochart
+    - SylvainCorlay
```
Comment (Anthchirp) [inline recipes/xtensor-zarr/meta.yaml:55]: still needs to confirm before merging

**Comment by davidbrochart** [comment]
@Anthchirp it should be fine now.

**Comment by Anthchirp** [comment]
Reverting the test hides a problem. Since you `exit 0` you stop at the first successful test, missing the (really failing) other one.

**Comment by davidbrochart** [comment]
Thanks a lot @Anthchirp, I have a bunch of other feedstocks to update now!

**Comment by Anthchirp** [comment]
Sorry 😉

**Pull request merged**